operators/pkg/forge/containers.go

```diff
@@ -0,0 +1,328 @@
+	probe := ContainerProbe()
+	probe.Handler.TCPSocket = &corev1.TCPSocketAction{
+		Port: intstr.FromString(portName),
+	}
+	c.ReadinessProbe = probe
+}
+
+// SetContainerReadinessHTTPProbe sets the given container's ReadinessProbe with a HTTPGet handler.
+func SetContainerReadinessHTTPProbe(c *corev1.Container, portName, path string) {
+	probe := ContainerProbe()
+	probe.Handler.HTTPGet = &corev1.HTTPGetAction{
+		Port: intstr.FromString(portName),
+		Path: path,
+	}
+	c.ReadinessProbe = probe
+}
+
+// ContainerProbe forges a Probe with the following values:
+// InitialDelaySeconds: 10
+// PeriodSeconds:       2
+// SuccessThreshold:    2
+// FailureThreshold:    5
+// It sets no handler.
+func ContainerProbe() *corev1.Probe {
+	return &corev1.Probe{
+		InitialDelaySeconds: 10,
+		PeriodSeconds:       2,
+		SuccessThreshold:    2,
+		FailureThreshold:    5,
+		Handler:             corev1.Handler{},
+	}
+}
+
+// SetContainerResources sets the given container's Resources with the given values:
+// cpu values are converted to millis: 1.2 -> 1200m; ram sizes are in MiB.
+func SetContainerResources(
+	c *corev1.Container,
+	cpuRequests, cpuLimits float32,
+	memRequestsMi, memLimitsMi int,
+) {
+	c.Resources = corev1.ResourceRequirements{
+		Requests: corev1.ResourceList{
+			corev1.ResourceCPU:    *resource.NewMilliQuantity(int64(cpuRequests*1e3), resource.DecimalSI),
+			corev1.ResourceMemory: *resource.NewQuantity(int64(memRequestsMi*(1<<20)), resource.BinarySI),
+		},
+		Limits: corev1.ResourceList{
+			corev1.ResourceCPU:    *resource.NewMilliQuantity(int64(cpuLimits*1e3), resource.DecimalSI),
+			corev1.ResourceMemory: *resource.NewQuantity(int64(memLimitsMi*(1<<20)), resource.BinarySI),
+		},
+	}
+}
+
+// SetContainerResourcesFromEnvironment sets the given container's Resources starting from the Resources specified inside the environment.
+func SetContainerResourcesFromEnvironment(c *corev1.Container, env *clv1alpha2.Environment) {
+	cpuRequestsMillis := env.Resources.CPU * env.Resources.ReservedCPUPercentage * 10
+
+	c.Resources = corev1.ResourceRequirements{
+		Requests: corev1.ResourceList{
+			corev1.ResourceCPU:    *resource.NewMilliQuantity(int64(cpuRequestsMillis), resource.DecimalSI),
+			corev1.ResourceMemory: env.Resources.Memory,
+		},
+		Limits: corev1.ResourceList{
+			corev1.ResourceCPU:    *resource.NewQuantity(int64(env.Resources.CPU), resource.DecimalSI),
+			corev1.ResourceMemory: env.Resources.Memory,
+		},
+	}
+}
+
+// ContainerVolume forges a Volume containing
+// a PVC source in case of persistent envs,
+// an emptydir in case of non persistent envs.
+func ContainerVolume(
+	volumeName, claimName string,
+	environment *clv1alpha2.Environment,
+) corev1.Volume {
+	volume := corev1.Volume{
+		Name:         volumeName,
+		VolumeSource: corev1.VolumeSource{},
+	}
+
+	if environment.Persistent {
+		volume.VolumeSource.PersistentVolumeClaim = &corev1.PersistentVolumeClaimVolumeSource{
+			ClaimName: claimName,
+		}
```

Comment:
Would directly create the volume and return it here,

Reply:
Even if I have to duplicate the general volume forging?

Reply:
You would duplicate only the name. Still, proceed as you prefer, it is fine from my side.